cpp/src/io/parquet/page_enc.cu

```diff
@@ -1315,56 +1315,44 @@ __global__ void __launch_bounds__(128, 8)
   }
 }
 
-// blockDim(128, 1, 1)
-__global__ void __launch_bounds__(128) gpuDecideCompression(device_span<EncColumnChunk> chunks)
+// blockDim(32, 1, 1)
+__global__ void __launch_bounds__(32) gpuDecideCompression(device_span<EncColumnChunk> chunks)
 {
-  // After changing the way structs are loaded from coop to normal, this kernel has no business
-  // being launched with 128 thread block. It can easily be a single warp.
   __shared__ __align__(8) EncColumnChunk ck_g;
-  __shared__ __align__(4) unsigned int error_count;
+  __shared__ __align__(4) unsigned int compression_error;
   using warp_reduce = cub::WarpReduce<uint32_t>;
   __shared__ typename warp_reduce::TempStorage temp_storage[2];
-  __shared__ volatile bool has_compression;
-
-  uint32_t t                      = threadIdx.x;
-  uint32_t uncompressed_data_size = 0;
-  uint32_t compressed_data_size   = 0;
-  uint32_t num_pages;
 
+  auto const t = threadIdx.x;
   if (t == 0) {
-    ck_g = chunks[blockIdx.x];
-    atomicAnd(&error_count, 0);
-    has_compression = false;
+    ck_g              = chunks[blockIdx.x];
+    compression_error = 0;
   }
-  __syncthreads();
-  if (t < 32) {
-    num_pages = ck_g.num_pages;
-    for (uint32_t page = t; page < num_pages; page += 32) {
-      auto& curr_page         = ck_g.pages[page];
-      uint32_t page_data_size = curr_page.max_data_size;
-      uncompressed_data_size += page_data_size;
-      if (auto comp_res = curr_page.comp_res; comp_res != nullptr) {
-        has_compression = true;
-        compressed_data_size += comp_res->bytes_written;
-        if (comp_res->status != compression_status::SUCCESS) { atomicAdd(&error_count, 1); }
-      }
+  __syncwarp();
```

Comment:
Can we write this kernel in a block-size agnostic way? Unlike `__syncthreads();`, using `__syncwarp();` assumes that `block_size == warp_size == 32`.

Reply:
That depends on how we would scale the parallelism with multiple warps. If any warps worked on a single `chunks` element, then, yes, we would need to syn all threads in the block. But, with multiple warps, IMO this kernel should actually have each warp would work on a separate `chunks` element. In this case we don't need to synchronize different warps and `__syncwarp` is still the right option.
I understand that my change left this ambiguous as warp size is used interchangeably for block and warp size. I'll try to make this clearer.

Reply:
Modified the kernel to work with any number of warps in a block. The size can be adjusted via `constexpr` `decide_compression_warps_in_block`. Used warp_size as well, so we should be magic number-free now :)